azure/services/identities/client.go

```diff
@@ -63,7 +64,7 @@ func (ac *AzureClient) GetClientID(ctx context.Context, providerID string) (stri
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "identities.GetClientID")
 	defer done()
 
-	parsed, err := arm.ParseResourceID(providerID)
+	parsed, err := arm.ParseResourceID(strings.TrimPrefix(providerID, "azure://"))
```

Comment:
Did you remove the other changes because they were not needed? in any case this should probably be a const as well

Do we have webhook validation to ensure the provided ID is in this format?

Reply:
I refactored it so the change is in one place rather than three. I'll take a look at webhook validation and a unit test for this func.

Reply:
I refactored this to be more general, so anywhere `arm.ParseResourceID()` was called, now we call a wrapper func that first removes the "azure://" prefix if it's present. I think this is a safer fix that prevents other similar problems.